semgrep/sol-rules.yaml

```diff
@@ -152,11 +153,15 @@ rules:
     severity: ERROR
     message: Use abi.encodeCall instead of abi.encodeWithSelector
     patterns:
-      - pattern: |
-          abi.encodeWithSelector(...);
+      - pattern-either:
+          - pattern: |
+              abi.encodeWithSelector(...);
+          - pattern: |
+              abi.encodeWithSignature(...);
+      - pattern-not: vm.expectRevert(abi.encodeWithSelector(...));
+      - pattern-not: vm.expectRevert(abi.encodeWithSignature(...));
```

Comment:
Let's remove this line, we actually want to force people to use the correct selector. I could only find one instance where this was used in the codebase (`vm.expectRevert(abi.encodeWithSignature("Panic(uint256)", 0x32));`) and this should actually be `vm.expectRevert(stdError.indexOOBError)` where `stdError` is imported from `forge-std`.